### DISCUSSION TITLE
getInitialNotification fire on simulator push (IOS)

### DISCUSSION THREAD

**Discussion by moh3n9595** [invertase/react-native-firebase#4675] [Ideas]
After Xcode 11.4 released, simulators support simulating remote push notifications, including background content fetch notifications. ([Release note](https://developer.apple.com/documentation/xcode-release-notes/xcode-11_4-release-notes))

It can be good if `react-native-firebase` works with manual APNS pushes.

**Comment by mikehardy** [comment]
Have you tried it? Please do so if you have not and report your results

More specifically: this has nothing to do with react-native-firebase supporting it or not, if the Simulator pretends it received an APNS push as you drop a JSON on it, then the firebase-ios-sdk will receive it. The only truly supported method of testing is on real devices though, as sometimes there are subtle differences in how the JSON goes through even, between Simulator and real device

**Reply by moh3n9595** [reply]
Yes, I tried it on the simulator and finally, I got a notification on the IOS simulator, but some functionality like click on the notification doesn't call methods like `getInitialNotification()`.

Maybe the problem is related to my APNS file data. Need some `firebaseId` or something like this on it? 
This is what I tested:
```
{
    "Simulator Target Bundle": "com.MY_APP",
    "aps": {
        "alert": "This is a simulated notification!",
        "badge": 3,
        "sound": "default"
    }
}
```

[Here](https://stackoverflow.com/questions/1080556/how-can-i-test-apple-push-notification-service-without-an-iphone/60085404#60085404) you can find a quick way to test it.

**Reply by mikehardy** [reply]
This is not something we officially support, so I won't attempt to troubleshoot it. But I knew it would work if you dropped the right JSON on the simulator and I'm happy to see you've got that working. While this may be a way of testing things informally / quickly, we only have reproducible success testing on real devices and that is what we work with in this repo

**Reply by Crysp** [reply]
Looks like APNS file data requires additional field `gcm.message_id`

https://github.com/invertase/react-native-firebase/blob/main/packages/messaging/ios/RNFBMessaging/RNFBMessaging%2BUNUserNotificationCenter.m#L155-L161

But notification with data below does not return via `getInitialNotification()`

```JSON
{
  "Simulator Target Bundle": "com.group.app",
  "gcm.message_id": "1234567800998877",
  "aps": {
    "alert": {
      "title": "Notification",
      "body": "You notified"
    }
  }
}
````

@mikehardy could you suggest what kind of data is required?
